docs/features/secrets/general.md

```diff
@@ -0,0 +1,52 @@
+---
+title: "Consuming Secrets"
+layout: default
+---
+
+## Consuming secrets
+Every provider implements `ISecretProvider` which makes it easy to use a consistent flow, regardless of the provider.
+
+You can easily retrieve secrets as following
+
+```csharp
+var secret = await secretProvider.Get("EventGrid-AuthKey");
+```
+
+## Caching Secrets
+Some secret providers recommend to cache secrets for a while to avoid hitting the service limitations.
+
+We provide a `CachedSecretProvider` which allows them to be cached in memory for a certain amount of time.
+
+```csharp
+var cachedSecretProvider = new CachedSecretProvider(secretProvider);
+var secret = await cachedSecretProvider.Get("EventGrid-AuthKey");
+```
+
+If you prefer a more fluent approach you can also use our `WithCaching` extension.
+
+```csharp
+var cachedSecretProvider = new KeyVaultSecretProvider(vaultAuthenticator, vaultConfiguration)
+                                    .WithCaching();
+var secret = await cachedSecretProvider.Get("EventGrid-AuthKey");
+```
+
+### Configuring the cache
+By default we only keep them around for 5 minutes, but you can configure this yourself.
```

Comment:
Not sure if we want to do that as changing the default would mean breaking contracts. _(Although you could say that changing default is also breaking)_

I *personally* would keep it like this because there is an overload which they can use if they want to be explicit.

Reply:
Yes, ok. Maybe set the default in bold then?

Reply:
Done!